OsmAnd-java/src/main/java/net/osmand/search/SearchUICore.java

```diff
@@ -852,7 +854,36 @@ public boolean publish(SearchResult object) {
 			}
 			return false;
 		}
-		
+
+		public List<String> getTranslatedAdditionalInfo(Amenity amenity) {
+			List<String> result = new ArrayList<>();
+			Map<String, String> additionalInfo = amenity.getInternalAdditionalInfoMap();
+			if (!Algorithms.isEmpty(additionalInfo)) {
+				MapPoiTypes poiTypes = MapPoiTypes.getDefault();
+				for (Map.Entry<String, String> entry : additionalInfo.entrySet()) {
+					String key = entry.getKey();
+					String value = entry.getValue();
+					AbstractPoiType pt = poiTypes.getAnyPoiAdditionalTypeByKey(key);
+					if (pt == null && !Algorithms.isEmpty(value) && value.length() < 50) {
```

Comment:
Is it a copy paste another method?

Reply:
It is partially copied from [here](https://github.com/osmandapp/OsmAnd/blob/08ea213514837408f0d68394d5081bc76a8e0979/OsmAnd/src/net/osmand/plus/helpers/AmenityExtensionsHelper.java#L93)  and no idea how to refactoring.
Better to left as is.

Reply:
No idea too but it's definitely not correct to just copy, this code is used for editing or for displaying context menu and it doesn't look logical to reuse in search. 
So code is not clear and I couldn't accept it.